File: Patrones Comportamiento/Mediator/Ejemplo.js
// Mediator
class Mediator {
  notify(message) {
    console.log(message);
  }
}

class User {
  constructor(mediator) {
    this.mediator = mediator;
  }
  send() {
    this.mediator.notify("Mensaje enviado");
  }
}

new User(new Mediator()).send();

File: Patrones Comportamiento/Mediator/Ejemplo.py
# Mediator
class Mediator:
    def notify(self, message):
        print(message)

class User:
    def __init__(self, mediator):
        self.mediator = mediator

    def send(self):
        self.mediator.notify("Mensaje enviado")

User(Mediator()).send()
